Templates/AppSource App/.AL-Go/localDevEnv.ps1

```diff
@@ -1,38 +1,109 @@
+                                                                
+'@
+
+Write-Host @'
+This script will create a docker based local development environment for your project.
+All apps and test apps will be compiled and published to the environment in the development scope.
+The script will also modify launch.json to have a Local Sandbox configuration point to your environment.
+
+'@
+
+$settings = ReadSettings -baseFolder $baseFolder -userName $env:USERNAME
+
+Write-Host "Checking System Requirements"
+$dockerService = Get-Service docker -ErrorAction SilentlyContinue
+if (-not $dockerService -or $dockerService.Status -ne "Running") {
+    throw "Creating a local development enviroment requires you to have Docker installed and running."
+}
+if ($settings.keyVaultName) {
+    if (-not (Get-Module -ListAvailable -Name 'Az.KeyVault')) {
+        throw "A keyvault name is defined in Settings, you need to have the Az.KeyVault PowerShell module installed or you can set the keyVaultName to an empty string in the user settings file ($($ENV:UserName).Settings.json)."
+    }
+}
+
+Write-Host
+
+if (-not $containerName) {
+    $containerName = Enter-Value `
+        -title "Container name" `
+        -question "Please enter the name of the container to create" `
+        -default "bcserver"
+}
+
+if (-not $auth) {
+    $auth = Select-Value `
+        -title "Authentication mechanism for container" `
+        -options @{ "Windows" = "Windows Authentication"; "UserPassword" = "Username/Password authentication" } `
+        -question "Select authentication mechanism for container" `
+        -default "UserPassword"
+}
+
+if (-not $credential) {
+    if ($auth -eq "Windows") {
+        $credential = Get-Credential -Message "Please enter your Windows Credentials" -UserName $env:USERNAME
+        $CurrentDomain = "LDAP://" + ([ADSI]"").distinguishedName
+        $domain = New-Object System.DirectoryServices.DirectoryEntry($CurrentDomain,$credential.UserName,$credential.GetNetworkCredential().password)
+        if ($null -eq $domain.name) {
+            Write-Host -ForegroundColor Red "Unable to verify your Windows Credentials, you might not be able to authenticate to your container"
+        }
+    }
+    else {
+        $credential = Get-Credential -Message "Please enter username and password for your container" -UserName "admin"
     }
 }
-else {
-    $credential = Get-Credential -Message "Please enter username and password for your container" -UserName "admin"
+
+if (-not $licenseFileUrl) {
+    if ($settings.type -eq "AppSource App") {
+        $description = "When developing AppSource Apps, your local development environment needs the developer licensefile with permissions to your AppSource app object IDs"
+    }
+    else {
+        $description = "When developing PTEs, your can optionally specify a developer licensefile with permissions to object IDs of your dependant apps"
+    }
+    $licenseFileUrl = Enter-Value `
```

Comment:
shouldnt it pass for pte deployment? as it is optional

Reply:
Good catch.
I have added a default for PTEs which results in no licensefile.
If you enter "none" as license file appsource apps you also get no license file - not sure anybody will use that, but the option is there.